modules/fedimint-ln-client/src/lib.rs

```diff
@@ -857,16 +889,22 @@ impl LightningClientModule {
         short_channel_id: u64,
         route_hints: Vec<fedimint_ln_common::route_hints::RouteHint>,
         network: Network,
+        dbtx: &mut ModuleDatabaseTransaction<'_>,
     ) -> anyhow::Result<(
         OperationId,
         Invoice,
         ClientOutput<LightningOutput, LightningClientStateMachines>,
     )> {
-        let payment_keypair = KeyPair::new(&self.secp, &mut rng);
+        let payment_keypair = self
+            .module_root_secret
+            .child_key(LIGHTNING_INCOMING_CHILD_ID)
+            .child_key(Self::get_next_incoming_contract_child_id(dbtx).await)
+            .to_secp_key(&self.secp);
+        // Use the public key of the `payment_keypair` as the preimage for the invoice.
         let preimage: [u8; 32] = payment_keypair.x_only_public_key().0.serialize();
         let payment_hash = bitcoin::secp256k1::hashes::sha256::Hash::hash(&preimage);
 
-        // Temporary lightning node pubkey
+        // Temporary lightning node pubkey - does not need to be deterministic
```

Comment:
Since this key just signs the invoice and doesn't handle anything with funds, I don't think this needs to be deterministic. LMK if I misinterpreted this.